format/Layout.md

```diff
@@ -583,6 +583,12 @@ even if the null bitmap of the parent union array indicates the slot is
 null.  Additionally, a child array may have a non-null slot even if
 the the types array indicates that a slot contains a different type at the index.
 
+## Dictionary encoding
```

Comment:
As an example, you could have the following data:

```
type: List<String>

[
 ['a', 'b'],
 ['a', 'b'],
 ['a', 'b'],
 ['c', 'd', 'e'],
 ['c', 'd', 'e'],
 ['c', 'd', 'e'],
 ['c', 'd', 'e'],
 ['a', 'b']
 ]

In dictionary-encoded form, this could appear as:

```
[0, 0, 0, 1, 1, 1, 0]

dictionary i

type: List<String>

[
 ['a', 'b'],
 ['c', 'd', 'e'],
]
```